### ISSUE TITLE
There should be a rule to disallow untyped function parameters but allow return type inference

### ISSUE THREAD

**Issue by ringeringeraja** [python/mypy#15799]
Feature

A `no-untyped-parameters` rule which will output an error if the signature of a function has untyped parameters but will allow inferring this function return type. It is somewhat how TypeScript's `noImplicityAny` currently works. Existing [function signature rules](https://mypy.readthedocs.io/en/stable/config_file.html#untyped-definitions-and-calls) will require parameters to be annotated but also the return type.

The following should give an error (parameter `a` isn't typed):
```python
def fn(a): ...
```

The following shouldn't, and the return type should be inferred as `int`:
```python
def fn(a: int): return a
```

**Comment by erictraut** [comment]
I think this is a duplicate of https://github.com/python/mypy/issues/10149.

**Comment by AlexWaygood** [comment]
> I think this is a duplicate of #10149.

yes
